File: typescript/0242-valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
  const mapS = mapLetters(s)
  const mapT = mapLetters(t)

  if (Object.keys(mapT).length !== Object.keys(mapS).length){
    return false
  }

  for (const mapSKey in mapS) {
    if(mapS[mapSKey] !== mapT[mapSKey]){
      return false
    }
  }

  return true
}

function mapLetters(s: string): {[key: string]: number}{
  const map: {[key: string]: number} = {}
  for (let i = 0; i < s.length; i++) {
    if (s[i] in map){
      map[s[i]]++
    }else{
      map[s[i]] = 1
    }
  }
  return map
}

File: csharp/0242-valid-anagram.cs
public partial class Solution {
    public bool IsAnagram(string s, string t) {
        var mapS = MapLetters(s);
        var mapT = MapLetters(t);

        if (mapT.Count != mapS.Count){
            return false;
        }

        foreach (var key in mapS.Keys) {
            if(!mapT.ContainsKey(key) || mapS[key] != mapT[key]){
                return false;
            }
        }

        return true;
    }

    private Dictionary<char, int> MapLetters(string s) {
        var map = new Dictionary<char, int>();
        foreach (var c in s) {
            if (map.ContainsKey(c)){
                map[c]++;
            } else {
                map[c] = 1;
            }
        }
        return map;
    }
}